docs/tutorials/web.md

```diff
@@ -5,10 +5,421 @@ has_children: false
+import _ from 'lodash';
+import {imagenetClasses} from '../data/imagenet';
+import { InferenceSession } from 'onnxruntime-web';
+
+const [predictions, inferenceTime] = await runSqueezenetModel(imageTensor);
+
+export async function runSqueezenetModel(preprocessedData: any): Promise<[any, number]> {
+    
+    // Create session and set options. See the docs here for more options: 
+    //https://onnxruntime.ai/docs/api/js/interfaces/InferenceSession.SessionOptions.html#graphOptimizationLevel
+
+    const session = await ort.InferenceSession
+                            .create('./_next/static/chunks/pages/squeezenet1_1.onnx', 
+                            { executionProviders: ['webgl'], graphOptimizationLevel: 'all' });
+    console.log('Inference session created')
+
+    // Run inference and get results.
+    var [results, inferenceTime] =  await runInference(session, preprocessedData);
+    return [results, inferenceTime];
+ 
+  }
+
+```
+
+Then let's call the `runInference` function by sending in the `session` and our input tensor `preprocessedData`.
+
+```javascript
+  async function runInference(session: InferenceSession, preprocessedData: any): Promise<[any, number]> {
+
+    // Get start time to calculate inference time.
+    const start = new Date();
+
+    // create feeds with the input name from model export and the preprocessed data.
+    const feeds: Record<string, ort.Tensor> = {};
+    feeds[session.inputNames[0]] = preprocessedData;
+    
+    // run the session inference.
+    const outputData = await session.run(feeds);
+
+    // Get the end time to calculate inference time.
+    const end = new Date();
+
+    // Convert to seconds.
+    const inferenceTime = (end.getTime() - start.getTime())/1000;
+
+    // Get output results with the output name from the model export.
+    const output = outputData[session.outputNames[0]];
+
+    //Get the softmax of the output data. The softmax transforms values to be between 0 and 1
+    var outputSoftmax = softmax(Array.prototype.slice.call(output.data))
+    
+    //Get the top 5 results.
+    var results = imagenetClassesTopK(outputSoftmax, 5);
+    console.log('results: ', results);
+    return [results, inferenceTime];
+  }
+  ```
+
+Once the inference completes, we return the top 5 results and time it took to run the inference. This is then displayed on the `ImageCanvas` web component.
+
+## The `data` Folder
+{: .no_toc }
+The data folder in this template has `imagenetClasses` that is used to assign the label based on the inferencing result index. Additionally, there is a `sample-image-urls.ts` provided for testing the application.
+
+## ImageCanvas FSX Element Web Component
+
+The `ImageCanvas.tsx` web component has the button and display elements. Below is the logic for the web component:
+
+```javascript
+
+import {useRef, useState} from 'react';
+import {IMAGE_URLS} from '../data/sample-image-urls';
+import { inferenceSqueezenet } from '../utils/predict'
```

Comment:
{InferenceSqueezenet} for consistent and ending ';'

Reply:
fixed